robottelo/ui/gpgkey.py

```diff
@@ -57,82 +57,62 @@ def delete(self, name, really=True):
-            element.click()
-            self.wait_for_ajax()
-            if product:
-                self.click(tab_locators["gpgkey.tab_products"])
-            else:
-                self.click(tab_locators["gpgkey.tab_repos"])
-            if self.wait_until_element(locators["gpgkey.product_repo"]):
-                element = self.find_element(
-                    locators["gpgkey.product_repo"]).get_attribute('innerHTML')
-                string = element.strip(' \t\n\r')
-                return string
+        self.click(element)
+        if product:
+            self.click(tab_locators['gpgkey.tab_products'])
         else:
-            raise UIError(
-                'Could not search the given gpg key "{0}"'.format(key_name)
-            )
+            self.click(tab_locators['gpgkey.tab_repos'])
+        if self.wait_until_element(locators['gpgkey.product_repo']):
+            element = self.find_element(
+                locators['gpgkey.product_repo']).get_attribute('innerHTML')
+            return element.strip(' \t\n\r')
 
     def assert_key_from_product(self, name, prd_element, repo=None):
         """Assert the key association after deletion from product tab."""
-        if prd_element:
-            prd_element.click()
-            self.wait_for_ajax()
-            if repo is not None:
-                self.click(tab_locators["prd.tab_repos"])
-                strategy = locators["repo.select"][0]
-                value = locators["repo.select"][1]
-                self.click((strategy, value % repo))
-                self.click(locators["repo.gpg_key_edit"])
-                element = Select(
-                    self.find_element(locators["repo.gpg_key_update"])
-                ).first_selected_option.text
-                if element == '':
-                    return None
-                else:
-                    raise UIError(
-                        'GPGKey "{0}" is still assoc with selected repo'
-                        .format(name)
-                    )
-            else:
-                self.click(tab_locators["prd.tab_details"])
-                self.click(locators["prd.gpg_key_edit"])
-                element = Select(
-                    self.find_element(locators["prd.gpg_key_update"])
-                ).first_selected_option.text
-                if element == '':
-                    return None
-                else:
-                    raise UIError(
-                        'GPG key "{0}" is still assoc with product'
-                        .format(name)
-                    )
+        self.click(prd_element)
+        if repo is not None:
+            self.click(tab_locators['prd.tab_repos'])
+            strategy, value = locators['repo.select']
+            self.click((strategy, value % repo))
+            self.click(locators['repo.gpg_key_edit'])
+            element = Select(
+                self.find_element(locators['repo.gpg_key_update'])
+            ).first_selected_option.text
```

Comment:
It's a pity we aren't dropping selenium's Selects completely... I was thinking - how about we create some helper in the Base class, name it smth like `def fetch_select_value()` and support both old & new dropdowns, so  we could completely drop selenium's selects + drop a lot of `"entity.fetch_smth"` locators?